x/oracle/keeper/grpc_query.go

```diff
@@ -93,6 +93,70 @@ func (k Querier) Request(c context.Context, req *types.QueryRequestRequest) (*ty
 	return &types.QueryRequestResponse{Request: &request, Reports: reports, Result: &result}, nil
 }
 
+func (k Querier) PendingRequests(c context.Context, req *types.QueryPendingRequestsRequest) (*types.QueryPendingRequestsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	valAddress, err := sdk.ValAddressFromBech32(req.ValidatorAddress)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("unable to parse given validator address: %v", err))
+	}
+
+	lastExpired := k.GetRequestLastExpired(ctx)
+	requestCount := k.GetRequestCount(ctx)
+
+	var pendingIDs []int64
+	for id := lastExpired + 1; int64(id) <= requestCount; id++ {
+		oracleReq := k.MustGetRequest(ctx, id)
+
+		// If all validators reported on this request, then skip it.
+		reports := k.GetReports(ctx, id)
+		if len(reports) == len(oracleReq.RequestedValidators) {
+			continue
+		}
+
+		// Skip if validator hasn't been assigned or has been reported.
+		if valAddress != nil {
```

Comment:
Why need to check it? It should be ok because we checking on line 102?

Reply:
Agreed. Fixed.